elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -0,0 +1,140 @@
+            AsyncQueryResult asyncQueryResult;
+            if (queryType.equals(QueryType.JSONAPI_V1_0)) {
+                MultivaluedMap<String, String> queryParams = getQueryParams(query);
+                response = elide.get(getPath(query), queryParams, user);
+                log.debug("JSONAPI_V1_0 getResponseCode: {}", response.getResponseCode());
+                log.debug("JSONAPI_V1_0 getBody: {}", response.getBody());
+            }
+            else if (queryType.equals(QueryType.GRAPHQL_V1_0)) {
+                response = runner.run(query, user);
+                log.debug("GRAPHQL_V1_0 getResponseCode: {}", response.getResponseCode());
+                log.debug("GRAPHQL_V1_0 getBody: {}", response.getBody());
+            }
+            if (response != null){
+                asyncQuery = asyncDbUtil.updateAsyncQuery(id, (asyncQueryObj) -> {
+                    asyncQueryObj.setStatus(QueryStatus.COMPLETE);
+                });
+
+                // Create AsyncQueryResult entry for AsyncQuery
+                asyncQueryResult = asyncDbUtil.createAsyncQueryResult(response.getResponseCode(), response.getBody(), asyncQuery, id);
+
+                // Add queryResult object to query object
+                asyncDbUtil.updateAsyncQuery(id, (asyncQueryObj) -> {
+                    asyncQueryObj.setResult(asyncQueryResult);
+                });
+            } else {
+                // If no response is returned on AsyncQuery request we set the QueryStatus to FAILURE
+                // No AsyncQueryResult will be set for this case
+                asyncQuery = asyncDbUtil.updateAsyncQuery(id, (asyncQueryObj) -> {
+                    asyncQueryObj.setStatus(QueryStatus.FAILURE);
+                 });
+            }
+        } catch (Exception e) {
+            log.error("Exception: {}", e.getMessage());
+            // If an Exception is encountered we set the QueryStatus to FAILURE
+            //No AsyncQueryResult will be set for this case
+            asyncDbUtil.updateAsyncQuery(id, (asyncQueryObj) -> {
+                asyncQueryObj.setStatus(QueryStatus.FAILURE);
+            });
+        }
+    }
+
+    /**
+     * This method parses the url and gets the query params and adds them into a MultivaluedMap
+     * to be used by underlying Elide.get method
+     * @param query query from the Async request
+     * @throws URISyntaxException URISyntaxException from malformed or incorrect URI
+     * @return MultivaluedMap with query parameters
+     */
+    protected MultivaluedMap<String, String> getQueryParams(String query) throws URISyntaxException {
+        URIBuilder uri;
+        uri = new URIBuilder(query);
+        MultivaluedMap<String, String> queryParams = new MultivaluedHashMap<String, String>();
+        for (NameValuePair queryParam : uri.getQueryParams()) {
+            queryParams.add(queryParam.getName(), queryParam.getValue());
+        }
+        log.debug("QueryParams: {}", queryParams);
```

Comment:
If we log the entire asyncQuery object, we won't need all of these separate, sprinkled log messages.